examples/view-integration/external-views/src/app.ts

```diff
@@ -3,42 +3,74 @@
  * Licensed under the MIT License.
  */
 
-import { StaticCodeLoader, TinyliciousModelLoader } from "@fluid-example/example-utils";
+import { StaticCodeLoader } from "@fluid-example/example-utils";
+import {
+	createDetachedContainer,
+	loadExistingContainer,
+} from "@fluidframework/container-loader/legacy";
+import { createRouterliciousDocumentServiceFactory } from "@fluidframework/routerlicious-driver/legacy";
+import {
+	createInsecureTinyliciousTestTokenProvider,
+	createInsecureTinyliciousTestUrlResolver,
+	createTinyliciousTestCreateNewRequest,
+} from "@fluidframework/tinylicious-driver/test-utils";
+import { createElement } from "react";
+// eslint-disable-next-line import/no-internal-modules
+import { createRoot } from "react-dom/client";
 
-import { DiceRollerContainerRuntimeFactory, IDiceRollerAppModel } from "./containerCode.js";
-import { renderDiceRoller } from "./view.js";
+import { DiceRollerContainerRuntimeFactory } from "./containerCode.js";
+import type { IDiceRoller } from "./interface.js";
+import { DiceRollerView } from "./view.js";
 
-/**
- * Start the app and render.
- *
- * @remarks We wrap this in an async function so we can await Fluid's async calls.
- */
-async function start() {
-	const tinyliciousModelLoader = new TinyliciousModelLoader<IDiceRollerAppModel>(
-		new StaticCodeLoader(new DiceRollerContainerRuntimeFactory()),
-	);
+const updateTabForId = (id: string) => {
+	// Update the URL with the actual ID
+	location.hash = id;
+
+	// Put the ID in the tab title
+	document.title = id;
+};
+
+const render = (diceRoller: IDiceRoller) => {
+	const appDiv = document.getElementById("app") as HTMLDivElement;
+	const appRoot = createRoot(appDiv);
+	appRoot.render(createElement(DiceRollerView, { diceRoller }));
+};
 
+const urlResolver = createInsecureTinyliciousTestUrlResolver();
+const tokenProvider = createInsecureTinyliciousTestTokenProvider();
+const documentServiceFactory = createRouterliciousDocumentServiceFactory(tokenProvider);
+const codeLoader = new StaticCodeLoader(new DiceRollerContainerRuntimeFactory());
+
+async function start(): Promise<void> {
 	let id: string;
-	let model: IDiceRollerAppModel;
+	let groceryList: IDiceRoller;
```

Comment:
I'm guessing this isn't supposed to be `groceryList` 😄

Reply:
Ah good catch :-D